.github/workflows/test.yml

```
@@ -0,0 +1,43 @@
+name: "Run Test and Package Latest JS"
+
+on:
+  push:
+    branches:
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - run: npm ci
+    - run: npm test
+
+  check-latest-dist:
+    runs-on: ubuntu-latest
+    needs: [unit-tests]
+    steps:
+    - uses: actions/checkout@v2
+    - run: npm install
+    - run: yarn prepare
+    - name: Check latest code is packaged in the dist folder
+      id: check-for-changes
+      run: |
+        if [ $(git status --porcelain | wc -l) -eq "0" ]; then
```

I reckon we could only check the status of `dist/index.js`, since if it's out-of-date it will have been modified by the previous step

---

```suggestion
        if [ -n "$(git status --porcelain -- dist/index.js)" ]
        then
```

needs quotes, I wouldn't quote an int passed to `-eq`, I prefer `then` on a newline but I might be in the minority

---

I'm going to extract the logic to other files to separate concerns and make testing easier in a follow up PR. The only way this can be different is if `yarn prepare` makes a change, which it'll only do in the `dist` folder, so I think this is an optimisation too far and isn't needed.

I'll update the `then` syntax, and unquote the int passed to `-eq`